aws_lambda_powertools/utilities/parser/parser.py

```diff
@@ -76,11 +77,20 @@ def handler(event: Order, context: LambdaContext):
     ValidationError
         When input event does not conform with model provided
     InvalidModelTypeError
-        When model given does not implement BaseModel
+        When model given does not implement BaseModel or is not provided
     InvalidEnvelopeError
         When envelope given does not implement BaseEnvelope
     """
-    parsed_event = parse(event=event, model=model, envelope=envelope) if envelope else parse(event=event, model=model)
+    if model is None:
+        type_hints = typing.get_type_hints(handler)
+        model = type_hints.get("event")
+        if model is None:
+            raise InvalidModelTypeError(
+                "The model must be provided either as the `model` argument to `event_parser`"
+                "or as the type hint of `event` in the handler that it wraps",
+            )
+
+    parsed_event = parse(event=event, model=model, envelope=envelope)
```

Comment:
It's better to keep the old version to avoid any problems with mypy. We have some overloads for the method `parse`

```suggestion
    parsed_event = parse(event=event, model=model, envelope=envelope) if envelope else parse(event=event, model=model)
```